### PR TITLE
feat: make a check on status change when you change the max amount of transactions

### PR DESCRIPTION
**Pull request by Vladimir-Petkov**
[AB#30200](https://dev.azure.com/redcrossnl/7d17f2a9-3f18-40f1-8c29-98568fbca7e4/_workitems/edit/30200)

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] I have added tests wherever relevant
- [x] I have made sure that all automated checks pass before requesting a review
- [x] I do not need any deviation from our PR guidelines

### CODE REVIEW

File: services/121-service/test/registrations/update-max-payment-registration.test.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    await resetDB(SeedScript.nlrcMultiple);
+    accessToken = await getAccessToken();
+
+    const registration = Object.assign({ maxPayments }, registrationPvScoped);
+    await seedPaidRegistrations([registration], programIdPv);
+  });
+
+  it('should succesfully update maxPayments without status change', async () => {
+    // Arrange
+    const reason = 'automated test';
```
[the remark this answers is not in the document]
Comment (Vladimir-Petkov) [reply services/121-service/test/registrations/update-max-payment-registration.test.ts]: Done.

File: services/121-service/test/registrations/update-max-payment-registration.test.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    // Assert
+    expect(response.statusCode).toBe(HttpStatus.OK);
+
+    const result = await searchRegistrationByReferenceId(
+      registrationPvScoped.referenceId,
+      programIdPv,
+      accessToken,
+    );
+    const registration = result.body.data[0];
+    expect(registration.maxPayments).toBe(maxPayments);
```
[the remark this answers is not in the document]
Comment (Vladimir-Petkov) [reply services/121-service/test/registrations/update-max-payment-registration.test.ts]: Done.

File: services/121-service/src/registration/registrations.service.ts [hunk 3]
```diff
@@ -514,6 +523,15 @@ export class RegistrationsService {
       }
     }
 
+    if (maxPaymentsMatchesPaymentCount) {
+      registrationToUpdate = await this.updateAttribute(
+        'registrationStatus',
+        'completed',
```
[the remark this answers is not in the document]
Comment (Vladimir-Petkov) [reply services/121-service/src/registration/registrations.service.ts]: Agree, and fixed.

File: services/121-service/test/registrations/update-max-payment-registration.test.ts [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+    const registration = result.body.data[0];
+
+    // Is old data still the same?
+    expect(registration.maxPayments).toBe(testPayments);
+  });
+
+  it('should succesfully update maxPayments and change status to completed', async () => {
+    // Arrange
+    const status = 'completed';
```
[the remark this answers is not in the document]
Comment (Vladimir-Petkov) [reply services/121-service/test/registrations/update-max-payment-registration.test.ts]: Agree.

File: services/121-service/test/registrations/update-max-payment-registration.test.ts [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+    const registration = Object.assign(
+      { maxPayments: testPayments },
+      registrationPvScoped,
+    );
+    await seedPaidRegistrations([registration], programIdPv);
+  });
+
+  it('should succesfully update maxPayments without status change', async () => {
+    // Arrange
+    const initialStatus = 'included';
```
[the remark this answers is not in the document]
Comment (Vladimir-Petkov) [reply services/121-service/test/registrations/update-max-payment-registration.test.ts]: Agree.

**Review by elwinschmitz — APPROVED** [review]

**Pull request merged**